### PR TITLE
[Decode] Fix memory out-of-bounds issue for VP8

### PR DESCRIPTION
**Pull request by Davis-HXX**
The data length exeding the max length of data.
Add checking of remaininga bitstream data size.
Add checking of n_bytes_offset size.
Add checking of dataSize size.

### CODE REVIEW

File: _studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp [hunk 1]
```diff
@@ -1003,309 +1003,312 @@ mfxStatus VideoDECODEVP8_HW::DecodeFrameHeader(mfxBitstream *in)
...
+        if (partitions > 1)
         {
-            m_frame_info.partitionSize[i] = (int32_t)(pTokenPartition[0]) |
-                                             (pTokenPartition[1] << 8) |
-                                             (pTokenPartition[2] << 16);
-            pTokenPartition += 3;
+            m_frame_info.partitionStart[0] = pTokenPartition + (partitions - 1) * 3;
+
+            if (m_frame_info.partitionStart[0] > data_in_end)
+                MFX_RETURN(MFX_ERR_MORE_DATA);
```
Comment (Davis-HXX) [inline _studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp:1182]: Out of bounds when reading pTokenPartion data from bitstream data.
Add checking of remaining bitstream data size.

File: _studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_common.cpp [hunk 2]
```diff
@@ -61,6 +61,10 @@ namespace VP8DecodeCommon
                 start_code_found = true;
                 // move to frame tag
                 p_bitstream -= 3;
+
+                if (n_bytes_offset < 3)
+                    MFX_RETURN(MFX_ERR_INVALID_VIDEO_PARAM);
+
```
Comment (Davis-HXX) [inline _studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_common.cpp:67]: Add checking of n_bytes_offset size.

File: _studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp [hunk 3]
```diff
@@ -1003,309 +1003,313 @@ mfxStatus VideoDECODEVP8_HW::DecodeFrameHeader(mfxBitstream *in)
     MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_HOTSPOTS, "VideoDECODEVP8_HW::DecodeFrameHeader");
     using namespace VP8Defs;
 
-    mfxU8* data_in     = 0;
-    mfxU8* data_in_end = 0;
-    mfxU8  version;
-    int width       = 0;
-    int height      = 0;
+    try{
```
Comment (Davis-HXX) [inline _studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp:1006]: Add checking of remaining bitstream data size.

File: _studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp [hunk 4]
```diff
@@ -1321,6 +1325,12 @@ mfxStatus VideoDECODEVP8_HW::DecodeFrameHeader(mfxBitstream *in)
     int fix = (m_boolDecoder[VP8_FIRST_PARTITION].bitcount() & 0x7) ? 1 : 0;
     m_frame_info.firstPartitionSize = m_frame_info.firstPartitionSize - (m_boolDecoder[VP8_FIRST_PARTITION].pos() - 3 + fix);
 #endif
+    }
+    catch(const vp8_exception & ex)
```
Comment (Davis-HXX) [inline _studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp:1329]: Add checking of remaining bitstream data size.

File: _studio/mfx_lib/decode/vp8/include/mfx_vp8_dec_decode_hw.h [hunk 5]
```diff
@@ -122,6 +122,10 @@ class MFX_VP8_BoolDecoder
         m_range = 255;
         m_bitcount = 8;
         m_pos = 0;
+
+        if (dataSize < 4)
```
Comment (Davis-HXX) [inline _studio/mfx_lib/decode/vp8/include/mfx_vp8_dec_decode_hw.h:126]: Add checking of dataSize size.

**Review by wangyan-intel — APPROVED** [review]

**Pull request merged**